plugins/Directory/src/Model/Table/DirectoriesTable.php

```diff
@@ -146,6 +146,13 @@ public function beforeDelete(Event $event, Entity $entity)
             $this->Alert->error($message, ['type' => 'string', 'reset' => true]);
             return false;
         }
+        $userId = $this->Session->read('Auth.User.id');
+        if($entity->id == $userId){
+            $event->stopPropagation();
+            $message = __('You cannot delete your record by your self');
```

Comment:
message is confusing a bit. 
something like "You are not allowed to delete your own profile" would be more meaningful